### DISCUSSION TITLE
Best practice for tensor containing time information?

### DISCUSSION THREAD

**Discussion by Bibi** [cornellius-gp/gpytorch#1768] [Q&A] [answered]
Hi,

I am using Gaussian process regression on time series (sampled at 10 min), with time as the input** and the value of signal as output. Using GPML on Matlab, I would use serial date number (datenum). Using `GPyTorch`, what would be the best way to define a tensor containing time information?

I currently use `read_csv` from `pandas` to import the csv data, then convert `datetime64` into `float64` with `timestamp()`, but I'm not sure that's the proper way to do it.

Thanks!

** Using indexes as input is not a good idea, since data is not regularly sampled and the signal exhibits some kind of periodic behaviour.

**Comment by wjmaddox** [comment, marked as answer]
Yes, raw floats is probably okay, although you'd probably want to do something like normalize them to [0, 1] based off of your minimum / maximum time (for numerical stability reasons).
